File: Contest/CarWale/1__.cpp
#include<bits/stdc++.h>

int getMinimumOperation(int items_count, int* items){
    int operation_count = 0;
    len(items)
    for(int i=0;i< items_count;i++){
        if (items[i] % 2 != items[i + 1] % 2){
            continue;
        }
        int num = items[i + 1]
        if (num % 2 == 0){
            while num & 1 == 0{
                num = num / 2;
                operation_count++;
            }
        }
        else{
            while num & 1 == 1{
                num = num / 2;
                operation_count++;
            }
        }
        items[i + 1] = num;
    }
    return operation_count;
}

int main(){
    int items_count = 5;
    int items = {4, 10, 10, 6, 2};
    int result = getMinimumOperation(items_count, items);
    return 0;
}

File: Contest/CarWale/1__.py

from itertools import count


def make_num_odd(num):
    count_ = 0
    if num&1 == 1:
        # making odd to even 
        while num & 1 == 1:
            num = int(num / 2)
            count_ += 1
    else:
        # making even to odd
        while num & 1 != 1:
            num = int(num / 2)
            count_ += 1
    return count_, num
def get_minimum_operation(items_count, items):
    operation_count = 0
    for i in range(len(items) - 1):
        print(items[i], items[i + 1], items[i] % 2 , items[i + 1] % 2)
        if items[i] % 2 != items[i + 1] % 2:
            continue
        count_, num = make_num_odd(items[i + 1])
        operation_count += count_
        items[i + 1] = num
    print(items)
    return operation_count



items = [4, 10, 10, 6, 2]
items = [6, 5, 9, 7, 3]
items = [4, 10, 10, 6, 3]
print(get_minimum_operation(0, items))